docs/api/design.md

```diff
@@ -0,0 +1,528 @@
+    id: int = Field(default=None, primary_key=True)
+    episode_id: str = Field(foreign_key="episode.uuid")
+    file_path: str
+    is_annotated: bool = False
+    created_at: datetime = Field(default_factory=datetime.now)
+    updated_at: datetime = Field(default_factory=datetime.now)
+    
+    episode: Episode = Relationship(back_populates="transcriptions")
+
+class Chapter(SQLModel, table=True):
+    """Represents a chapter marker in the episode"""
+    id: int = Field(default=None, primary_key=True)
+    episode_id: str = Field(foreign_key="episode.uuid")
+    title: str
+    start_time: float  # in seconds
+    end_time: float  # in seconds
+    level: int
+    
+    episode: Episode = Relationship()
+
+class MetadataFile(SQLModel, table=True):
+    """Represents metadata files such as chapter lists and show notes"""
+    id: int = Field(default=None, primary_key=True)
+    episode_id: str = Field(foreign_key="episode.uuid")
+    file_path: str
+    file_type: str  # "chapters", "show_notes"
+    created_at: datetime = Field(default_factory=datetime.now)
+    
+    episode: Episode = Relationship(back_populates="metadata_files")
+
+class ProcessingJob(SQLModel, table=True):
+    """Represents an asynchronous processing job"""
+    id: int = Field(default=None, primary_key=True)
+    episode_id: str = Field(foreign_key="episode.uuid")
+    job_type: str  # "preprocess", "postprocess", "metadata", "export", "audio_edit"
+    status: str  # "pending", "processing", "completed", "failed"
+    created_at: datetime = Field(default_factory=datetime.now)
+    updated_at: datetime = Field(default_factory=datetime.now)
+    error_message: Optional[str] = None
+
+# Request/Response Models
+
+class EditorStateUpdate(SQLModel):
+    """Model for updating editor state"""
+    editor_state: str
+
+class EditorStateResponse(SQLModel):
+    """Model for returning editor state"""
+    editor_state: str
+
+class PostProcessRequest(SQLModel):
+    """Model for initiating post-processing"""
+    whisper_options: Optional[str] = None
+    no_noise_remove: Optional[bool] = False
+
+class MetadataGenerationRequest(SQLModel):
+    """Model for initiating metadata generation"""
+    audio_file_id: Optional[int] = None
+
+class ExportRequest(SQLModel):
+    """Model for initiating export"""
+    embed_chapters: Optional[bool] = True
+    include_show_notes: Optional[bool] = True
+    format: Optional[str] = "mp3"  # "mp3", "wav"
+
+class AnnotatedSrtUpdateRequest(SQLModel):
+    """Model for updating annotated SRT"""
+    content: str
+
+class AnnotatedSrtUpdateResponse(SQLModel):
+    """Model for response after updating annotated SRT"""
+    success: bool
+    id: int
+    episode_id: str
+    is_annotated: bool
+    updated_at: datetime
+
+class JobStatusResponse(SQLModel):
+    """Model for job status information"""
+    id: int
+    episode_id: str
+    job_type: str
+    status: str
```

Comment:
this also should be a string of literals

Reply:
Same as above